src/modules/core/components/time-field/time-field.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  type FC,
+  type ChangeEvent,
+  useCallback,
+  useRef,
+  useState,
+  useEffect,
+} from 'react';
+import { useBoolean, useUpdateEffect } from 'usehooks-ts';
+// components
+import { TextField, TimeSelect } from '@/modules/core/components';
+import { Popover } from '@/modules/core/components/popover';
+// hooks
+import { useOnClickOutsideExtended } from '@/modules/core/hooks/use-on-click-outside-extended';
+// utils
+import { Time } from '@/modules/core/utils/time.utils';
+
+import type { TimeFieldProps } from './time-field.interface';
+import styles from './time-field.module.scss';
+
+export const TimeField: FC<TimeFieldProps> = ({
```

Comment:
Keyboard accessability is broken for this component. I can't use it via keyboard.

Reply:
How do you want to use it via keyboard? With keyboard you can change text-field...

Reply:
yep, but it's weird as you can actually use keyboard buttons up and down if you click inside the popover manually and it feels weird. We may need to discuss keyboard usage of this component more.